source/CMakeLists.txt

```diff
@@ -51,26 +51,26 @@ file(READ ${PROJECT_SOURCE_DIR}/config/MODEL_VER MODEL_VERSION)
 string(REPLACE "\n" " " MODEL_VERSION ${MODEL_VERSION})
 message(STATUS "Supported model version: ${MODEL_VERSION}")
 
-# define USE_CUDA_TOOLKIT
-if (DEFINED USE_CUDA_TOOLKIT)
-  if (USE_CUDA_TOOLKIT)
-    find_package(CUDA REQUIRED)
-  else()
-    message(STATUS "Will not build nv GPU support")
-  endif()
-else()
-  find_package(CUDA QUIET)  
-  if (CUDA_FOUND)
-    set(USE_CUDA_TOOLKIT TRUE)
-    message(STATUS "Found CUDA in ${CUDA_TOOLKIT_ROOT_DIR}, build nv GPU support")
-  else()
-    set(USE_CUDA_TOOLKIT FALSE)
-    message(STATUS "No cuda support found, will not build nv GPU support")
-  endif()
+# Devices that have both ROCM and CUDA are not currently supported
+if (USE_ROCM_TOOLKIT AND USE_CUDA_TOOLKIT)
+  message (FATAL_ERROR "Devices that have both ROCM and CUDA are not currently supported")
 endif()
+
+# define USE_CUDA_TOOLKIT
 if (USE_CUDA_TOOLKIT)
+  find_package(CUDA REQUIRED)
   add_definitions("-D GOOGLE_CUDA")
-endif()
+  message(STATUS "Found CUDA in ${CUDA_TOOLKIT_ROOT_DIR}, build nv GPU support")
+endif(USE_CUDA_TOOLKIT)
```

Comment:
else we should let the user know the user that we will not build with cuda support.